File: Intermediate/Topic-003-Introduction-to-Arrays/Homework/Q2-Time-to-equality/Time-to-equality.c
/**
 * @input A : Integer array
 * @input n1 : Integer array's ( A ) length
 * 
 * @Output Integer
 */

 int maximum(int *A, int n1) {
     int idx;
     int max = A[0];
     for ( idx = 1; idx < n1; idx++) {
         if (A[idx] > max) {
             max = A[idx];
         }

     }

     return max;
 }
int solve(int* A, int n1) {

    int max = maximum(A, n1);
    int idx, time;

    for (idx = 0; idx < n1; idx++) {
        time += (max - A[idx]) ;
    }


    return time;

}

File: Intermediate/Topic-003-Introduction-to-Arrays/Homework/Q2-Time-to-equality/Time-to-equality.py
def maximum(A):
    max = A[0]
    for i in range(1, len(A)):
        if A[i] > max :
            max = A[i]
    return max

class Solution:
    # @param A : list of integers
    # @return an integer
    def solve(self, A):
        max = maximum(A)
        time = 0
        for i in range(len(A)):
            time += (max - A[i])
        
        return time
